lib/node-metrics.js

```diff
@@ -9,87 +10,88 @@ let intervalId;
-    },
-    {
-      MetricName: 'NodeMemoryHeapUsed',
-      Unit: 'Bytes',
-      Value: memoryStats.heapUsed,
-    },
-    {
-      MetricName: 'NodeMemoryExternal',
-      Unit: 'Bytes',
-      Value: memoryStats.external,
-    },
-    {
-      MetricName: 'NodeCpuUser',
-      Unit: 'Percent',
-      Value: userCpuPercent,
-    },
-    {
-      MetricName: 'NodeCpuSystem',
-      Unit: 'Percent',
-      Value: systemCpuPercent,
-    },
-    {
-      MetricName: 'NodeCpuTotal',
-      Unit: 'Percent',
-      Value: totalCpuPercent,
-    },
-  ];
+    const metrics = [
+      {
+        MetricName: 'NodeEventLoopDelay',
+        Unit: 'Milliseconds',
+        Value: loopbench.delay,
+      },
+      {
+        MetricName: 'NodeMemoryRss',
+        Unit: 'Bytes',
+        Value: memoryStats.rss,
+      },
+      {
+        MetricName: 'NodeMemoryHeapTotal',
+        Unit: 'Bytes',
+        Value: memoryStats.heapTotal,
+      },
+      {
+        MetricName: 'NodeMemoryHeapUsed',
+        Unit: 'Bytes',
+        Value: memoryStats.heapUsed,
+      },
+      {
+        MetricName: 'NodeMemoryExternal',
+        Unit: 'Bytes',
+        Value: memoryStats.external,
+      },
+      {
+        MetricName: 'NodeCpuUser',
+        Unit: 'Percent',
+        Value: userCpuPercent,
+      },
+      {
+        MetricName: 'NodeCpuSystem',
+        Unit: 'Percent',
+        Value: systemCpuPercent,
+      },
+      {
+        MetricName: 'NodeCpuTotal',
+        Unit: 'Percent',
+        Value: totalCpuPercent,
+      },
+    ];
 
-  const cloudwatch = new AWS.CloudWatch(config.awsServiceGlobalOptions);
-  /** @type {import('aws-sdk').CloudWatch.Types.Dimensions} */
-  const dimensions = [
-    { Name: 'Server Group', Value: config.groupName },
-    { Name: 'InstanceId', Value: `${config.instanceId}:${config.serverPort}` },
-  ];
-  /** @type {import('aws-sdk').CloudWatch.Types.PutMetricDataInput} */
-  const params = {
-    Namespace: 'PrairieLearn',
-    MetricData: metrics.map((m) => ({
-      ...m,
-      StorageResolution: 1,
-      Timestamp: new Date(),
-      Dimensions: dimensions,
-    })),
-  };
-  cloudwatch.putMetricData(params, (err) => {
-    if (err) {
-      logger.error('Error reporting Node metrics to CloudWatch', err);
-    }
-  });
+    const cloudwatch = new AWS.CloudWatch(config.awsServiceGlobalOptions);
+    /** @type {import('aws-sdk').CloudWatch.Types.Dimensions} */
+    const dimensions = [
+      { Name: 'Server Group', Value: config.groupName },
+      { Name: 'InstanceId', Value: `${config.instanceId}:${config.serverPort}` },
+    ];
+    await cloudwatch
+      .putMetricData({
+        Namespace: 'PrairieLearn',
+        MetricData: metrics.map((m) => ({
+          ...m,
+          StorageResolution: 1,
+          Timestamp: new Date(),
+          Dimensions: dimensions,
+        })),
+      })
+      .promise();
+  } catch (err) {
+    logger.error('Error reporting Node metrics', err);
+    Sentry.captureException(err);
```

Comment:
But I guess we'll soon start failing our health checks and someone will kill us off, so maybe it's no big deal either way.

Reply:
Hmm... I know that crashing here has indeed saved us during `EMFILE` issues. And the health checks won't help us on non-chunk servers. I'm torn because while the crashing was convenient, we don't treat *other* errors as crash-worthy.

Reply:
Yeah, I'm fine to not crash.